snd-server/recommendation_engine.py

```diff
@@ -43,3 +67,50 @@ def get_top_n(
             result.append(RecommendedTrack(id=row["id"], similarity=row["similarity"]))
 
         return result
+
+    def get_top_n(
+        self, track: SoundationsTrackWithFeatures, n: int
+    ) -> List[RecommendedTrack]:
+        return self.__recommend(dataset=self.dataset, track=track, n=n)
+
+    async def get_top_n_for_user(
+        self, track_description_prompt: str, user: UserInDB, n: int
+    ) -> List[RecommendedTrack]:
+        user_history = await self.ctx.search_history.get_search_history(user=user)
+        if not user_history:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail="User track search history not found",
+                headers={"WWW-Authenticate": "Bearer"},
+            )
```

Comment:
I would probably come up with some fallback flow if there is no history yet, but we can keep it for now